express/component_spec.py

```diff
@@ -0,0 +1,328 @@
+"""This module defines classes to represent an Express component specification."""
+import copy
+import json
+import yaml
+import pkgutil
+import types
+import typing as t
+from dataclasses import dataclass, asdict
+
+import jsonschema.exceptions
+from jsonschema import Draft4Validator
+
+from express.exceptions import InvalidComponentSpec
+from express.io_utils import load_yaml
+from express.common import Field
+from express.pipeline_utils import get_kubeflow_type
+
+
+@dataclass
+class Input:
+    """
+    Component input arguments
+    Attributes:
+        name: name of the argument
+        description: argument description
+        type: the python argument type (str, int, ...)
+    """
+
+    name: str
+    description: str
+    type: str
+
+
+@dataclass
+class Output:
+    """
+    Component output argument
+    Attributes:
+        name: name of the argument
+        description: argument description
+    """
+
+    name: str
+    description: str
+
+
+T = t.TypeVar("T", Input, Output)
```

Comment:
this looks very cryptic to me, makes the code pretty hard to read

I'd just use the longer names to make it clearer

Reply:
Ok but people will see a T and will wonder what it means

Reply:
T" is a common convention used in Python (and other programming languages) to represent a generic type parameter. 
https://peps.python.org/pep-0585/